### PR TITLE
feat(components): Refactor options as li elements

### PR DESCRIPTION
**Pull request by dlrussel**
## Motivations

Refactor to update the combobox options as li elements instead of radio inputs.
We made this decision due to accessibility concerns with the fact that when using a radio input when using the arrow keys to switch the selected value an option would be made without using spacebar or enter to confirm a selection and the combobox content would close automatically. This created a weird experience for the user.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by taylorvnoj** [comment]
LGTM - @ZakaryH if you wanna do the approval that would be lovely 😄

File: packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.css [hunk 1]
```diff
@@ -19,8 +19,19 @@
...
+.optionsList {
+  margin: 0;
+  padding: 0;
+  list-style: none;
+}
+
 .option {
-  display: block;
+  padding: var(--space-small) var(--space-base);
+  cursor: pointer;
```
Comment (ZakaryH) [inline packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.css:30]: we'll need to have this rule in `.option:hover` for it to have the desired effect, but we can make a note to do that in the styling PR, or we could do it now. I don't mind terribly either way!

**Review by ZakaryH — APPROVED** [review]
I'm sad we can't use `button` because that gets us the click & return/space bar activation for free. using `li` we will need to handle the keyboard events ourselves. I can do that in my accessibility PR to get this merged ASAP to unblock the other tasks. 

this LGTM - but maybe change the title of the PR since we're not using buttons now!

File: packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.css (hunk 1 above)
Comment (taylorvnoj) [reply packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.css:30]: I've got it in my styling PR 👍

**Pull request merged**